$NetBSD: patch-src_Makefile,v 1.8 2025/11/21 09:05:36 adam Exp $

Add DESTDIR support.
Fix NetBSD support.
Don't build module tests.

--- src/Makefile.orig	2025-11-18 14:58:24.000000000 +0000
+++ src/Makefile
@@ -68,8 +68,8 @@ else
 endif
 
 PREFIX?=/usr/local
-INSTALL_BIN=$(PREFIX)/bin
-INSTALL=install
+INSTALL_BIN=$(DESTDIR)$(PREFIX)/bin
+INSTALL=$(BSD_INSTALL_PROGRAM)
 PKG_CONFIG?=pkg-config
 
 ifndef PYTHON
@@ -213,6 +213,10 @@ ifeq ($(uname_S),NetBSD)
 	    FINAL_LIBS+= -lexecinfo
     	endif
 else
+ifeq ($(uname_S),NetBSD)
+	# NetBSD
+	FINAL_LIBS+= -lpthread
+else
 ifeq ($(uname_S),FreeBSD)
 	# FreeBSD
 	FINAL_LIBS+= -lpthread -lexecinfo
@@ -248,6 +252,7 @@ endif
 endif
 endif
 endif
+endif
 
 ifdef OPENSSL_PREFIX
 	OPENSSL_CFLAGS=-I$(OPENSSL_PREFIX)/include
@@ -391,7 +396,7 @@ REDIS_CHECK_RDB_NAME=redis-check-rdb$(PR
 REDIS_CHECK_AOF_NAME=redis-check-aof$(PROG_SUFFIX)
 ALL_SOURCES=$(sort $(patsubst %.o,%.c,$(REDIS_SERVER_OBJ) $(REDIS_VEC_SETS_OBJ) $(REDIS_CLI_OBJ) $(REDIS_BENCHMARK_OBJ)))
 
-all: $(REDIS_SERVER_NAME) $(REDIS_SENTINEL_NAME) $(REDIS_CLI_NAME) $(REDIS_BENCHMARK_NAME) $(REDIS_CHECK_RDB_NAME) $(REDIS_CHECK_AOF_NAME) $(TLS_MODULE) module_tests
+all: $(REDIS_SERVER_NAME) $(REDIS_SENTINEL_NAME) $(REDIS_CLI_NAME) $(REDIS_BENCHMARK_NAME) $(REDIS_CHECK_RDB_NAME) $(REDIS_CHECK_AOF_NAME) $(TLS_MODULE)
 	@echo ""
 	@echo "Hint: It's a good idea to run 'make test' ;)"
 	@echo ""
